clutch/demo/alpha.lurk

```diff
@@ -121,7 +124,36 @@
 
 
 !(:commit (fn<-db ledger))
-0x36283c9973a837574e3e5a15815b60a541df265b0f218ab048e99a7c44ee3769
 
 !(:call 0x36283c9973a837574e3e5a15815b60a541df265b0f218ab048e99a7c44ee3769 '(1 0 2))
-;; !(:prove)
+!(:prove)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Chain
+
+!(:def chain<-db (lambda (db secret)
+                   (letrec ((foo (lambda (state msg)
+                                   (let ((new-state ((fn<-db state) msg)))
+                                     (cons new-state (hide secret (foo new-state)))))))
+                     (foo db))))
+
+!(:commit (chain<-db ledger 999))
+
+!(:chain  0x0757a47095c97d0a58a8ff66a2146949ee9d60d3f0a82887c203edf1748be711 '(1 0 2))
+;; (comm 0x14f7a28206656cb7cd6bfcdda773c470653f3136d894d9cb8240e28e3eb52e55)
+;; ((:ID 0 :BALANCE 122 :LAST-NAME "Church" :FIRST-NAME "Alonzo") (:FIRST-NAME "Alan" :LAST-NAME "Turing" :BALANCE 456 :ID 1) (:ID 2 :BALANCE 9001 :LAST-NAME "Nakamoto" :FIRST-NAME "Satoshi"))
+
+
+; !(:prove)
```

Comment:
This one needs a proof. To make it, edit the source to uncomment the line (by removing the initial `;`), then step through the demo (by hitting return) until you get to this line. Then hit return one more time and wait for it to return. It could take a long time.